lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+            return string
+        else:
+            return None
+
+    def cast_to_integer(string: str) -> Optional[int]:
+        try:
+            return int(string)
+        except ValueError:
+            return None
+
+    def cast_to_boolean(string: str) -> Optional[bool]:
+        if (string and string.lower() == 'yes') or string == '1':
+            return True
+        elif (string and string.lower() == 'no') or string == '0':
+            return False
+        return None
+
+    def build_response_answers(response: Union[str, List]) -> List:
+        answers = []
+        if not isinstance(response, list):
+            response = [response]
+
+        for item in response:
+            type_casted_item = casting_functions[response_type](item)
+
+            # cast_to_boolean can return False, so must be `is not None`
+            if type_casted_item is not None:
+                answers.append({ response_type: type_casted_item })
+
+        return answers
+
+    casting_functions: Mapping[str, Callable[[str], Any]] = {
+        'valueCoding': cast_to_coding,
+        'valueInteger': cast_to_integer,
+        'valueBoolean': cast_to_boolean,
+        'valueString': cast_to_string,
+        'valueDate': cast_to_string,
+    }
+
+    answers = build_response_answers(response)
+    if answers:
+        return create_questionnaire_response_item(question_id, answers)
+
+    return None
+
+
+def create_follow_up_encounter(record: REDCapRecord,
+                               patient_reference: dict,
+                               site_reference: dict,
+                               initial_encounter_reference: dict) -> tuple:
+    """
+    Returns a FHIR Encounter resource entry and reference for a follow-up
+    encounter
+    """
+    if not record.get('fu_timestamp'):
+        return None, None
+
+    encounter_identifier = create_identifier(
+        system = f"{INTERNAL_SYSTEM}/encounter",
+        value = f"{REDCAP_URL}{record.project.id}/{record['record_id']}/{record['redcap_event_name']}/{record['redcap_repeat_instance']}_follow_up"
+    )
+    encounter_class_coding = create_coding(
+        system = "http://terminology.hl7.org/CodeSystem/v3-ActCode",
+        code = "HH"
+    )
+    encounter_reason_code = create_codeable_concept(
+        system = "http://snomed.info/sct",
+        code = "390906007",
+        display = "Follow-up encounter"
+    )
+
+    # YYYY-MM-DD HH:MM in REDCap
+    encounter_date = record['fu_timestamp'].split()[0]
+    encounter_resource = create_encounter_resource(
+        encounter_identifier = [encounter_identifier],
+        encounter_class = encounter_class_coding,
+        encounter_date = encounter_date,
+        patient_reference = patient_reference,
+        location_references = [site_reference],
+        reason_code = [encounter_reason_code],
+        part_of = initial_encounter_reference
+    )
+
+    return create_entry_and_reference(encounter_resource, "Encounter")
+
```

Comment:
This function is an exact duplicate of the one in SCAN and should be imported from a shared module.

Reply:
Yes. The two ETLs should get refactored to use common code.